test/should_pass/try.erl

```diff
@@ -1,11 +1,11 @@
 -module('try').
 
--compile([export_all]).
+-compile([export_all, nowarn_export_all, nowarn_unused_vars]).
 
 -spec t() -> 2..4.
 t()->
     try A = 1 of
-        ok ->
+        1 ->
             %% a:b(A), unsafe
```

Comment:
not related to this PR, just this part reminded me that `A` here will be safe from OTP 24 (https://github.com/erlang/otp/pull/2662) so will have to update try/catch var env handling (admittedly quite an edge case)

Reply:
Ah, I didn't know that. I've created  #310 to track the issue.